ledger/participant-integration-api/src/main/scala/platform/store/backend/common/EventStorageBackendTemplate.scala

```diff
@@ -808,6 +808,18 @@ abstract class EventStorageBackendTemplate(
        """.as(get[Long](1).singleOpt)(connection)
   }
 
+  override def offsetAfter(start: Offset, after: Int)(connection: Connection): Option[Offset] =
+    SQL"""
+        WITH next_offsets_chunk AS (
+            SELECT event_offset
+            FROM participant_transaction_meta
+            ${QueryStrategy.whereOffsetHigherThanClause("event_offset", start)}
+            ORDER BY event_offset ASC
+            ${QueryStrategy.limitClause(Some(after))}
+         )
+       SELECT MAX(event_offset) AS max_offset_in_window FROM next_offsets_chunk"""
+      .as(offset("max_offset_in_window").?.single)(connection)
```

Comment:
This simplification is optimistic since it jumps over rejected commands. In cases where the ledger is polluted with high density of command rejections, the pruning window sizing can vary widely.

Reply:
> My gut-feel is that perhaps we should do event_sequential_id based windows. Does seem to be a more appropriate proxy, since they also incorporate the number of events which might wildly vary as well, and which is closer to the real pruning work what needs to be done. (also event-seq-id is numeric, so we might be already good off with guessing a fix window at the beginning and keep doing that)

It might be more precise, but is it easier? I didn't consider it, thinking that this solution is the fastest, simplest and a good enough proxy to what Canton is doing (Canton sets windows per offsets granularity, not events). of course, once each iteration is not O(ACS_size).. 🤦